### PR TITLE
Log errors in Traceloop Integration

### PR DESCRIPTION
**Pull request by nirga**
## Type

🐛 Bug Fix

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locall
If UI changes, send a screenshot/GIF of working UI fixes

### CODE REVIEW

File: litellm/integrations/traceloop.py [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
     def __init__(self):
-        from traceloop.sdk.tracing.tracing import TracerWrapper
-        from traceloop.sdk import Traceloop
+        try:
+            from traceloop.sdk.tracing.tracing import TracerWrapper
+            from traceloop.sdk import Traceloop
+
+            Traceloop.init(app_name="Litellm-Server", disable_batch=True)
+            self.tracer_wrapper = TracerWrapper()
+        except Exception as e:
```
Comment (paneru-rajan) [inline litellm/integrations/traceloop.py]: We should avoid catching the generic exception instead if we wanted to warn and suggest package is missing we should use ModuleNotFoundError exception.
```
ModuleNotFoundError: No module named 'traceloop'
```

File: litellm/integrations/traceloop.py [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
+            raise Exception(
+                f"\033[Traceloop not installed, try running 'pip install traceloop-sdk' to fix this error: {e}\n{traceback.format_exc()}\033[0m"
+            )
+
+    def log_error(kwargs, response_obj, start_time, end_time):
```
Comment (paneru-rajan) [inline litellm/integrations/traceloop.py]: `log_error` method is not in use. The addition of this is unclear.

File: litellm/integrations/traceloop.py [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
-        from traceloop.sdk import Traceloop
+        try:
+            from traceloop.sdk.tracing.tracing import TracerWrapper
+            from traceloop.sdk import Traceloop
+
+            Traceloop.init(app_name="Litellm-Server", disable_batch=True)
+            self.tracer_wrapper = TracerWrapper()
+        except Exception as e:
+            raise Exception(
+                f"\033[Traceloop not installed, try running 'pip install traceloop-sdk' to fix this error: {e}\n{traceback.format_exc()}\033[0m"
```
Comment (paneru-rajan) [inline litellm/integrations/traceloop.py]: You are missing `91m` characters before `Traceloop` in the error message causing the color to be white and missing `T`, check the image
image

File: litellm/integrations/traceloop.py [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
+
 class TraceloopLogger:
     def __init__(self):
-        from traceloop.sdk.tracing.tracing import TracerWrapper
-        from traceloop.sdk import Traceloop
+        try:
+            from traceloop.sdk.tracing.tracing import TracerWrapper
+            from traceloop.sdk import Traceloop
+
+            Traceloop.init(app_name="Litellm-Server", disable_batch=True)
```
Comment (paneru-rajan) [inline litellm/integrations/traceloop.py]: If we are tryin to catch and present a different message in the console if there is no traceloop package, adding any extra statement inside the try:except might confuse user when exception occurs from those extra statements.
```python
Traceloop.init(app_name="Litellm-Server", disable_batch=True)
self.tracer_wrapper = TracerWrapper()
```

**Review by paneru-rajan — COMMENTED** [review]
Hi @nirga, thank you for this changes, however, i have concerns around similar pattern i see everywhere.

I see the way we are catching Generic Exception and Throwing Generic Exception, which is not a recommended approach. While dealing with exception always a good idea to handle specific exception that we actually wanted.

I rather suggest something like:
```python
In [37]: try:
    ...:     from traceloop.sdk.tracing.tracing import TracerWrapper
    ...:     from traceloop.sdk import Traceloop
    ...: except ModuleNotFoundError as e:
    ...:     raise ModuleNotFoundError(f"\033[91mTraceloop not installed, try running 'pip install traceloop-sdk' to fix this error: {e}\033[0m")
```
Which will give something like in the console, you can see traceback is present by default no need to add that again:
image

If we use logging module, `logging.exception` is perfect candidate for this scenario.


CC: @ishaan-jaff / @krrishdholakia

File: litellm/integrations/traceloop.py (hunk 2 above)
Comment (nirga) [reply litellm/integrations/traceloop.py]: @paneru-rajan thanks, so how do I log errors from litellm?

Comment (ghost) [reply litellm/integrations/traceloop.py]: add it here - https://github.com/BerriAI/litellm/blob/5b1f9283412661c9abf028982523f14c172f8195/litellm/utils.py#L2437

Comment (nirga) [reply litellm/integrations/traceloop.py]: Done! @krrishdholakia

**Comment by ghost** [comment]
Hey @nirga thanks for this! can you just share a screenshot of this passing test_traceloop.py 

Should be good to merge then

**Comment by nirga** [comment]
@krrishdholakia here
Screenshot 2024-05-25 at 0 19 41

**Comment by ghost** [comment]
LGTM!

**Pull request merged**